### PR TITLE
log filtered examples

### PR DESCRIPTION
**Pull request by JanekEbb**

### CODE REVIEW

File: lazy_dataset/core.py [hunk 1]
```diff
@@ -1771,14 +1773,20 @@ def __getitem__(self, item):
...
             except self.exceptions as e:
+                catched_count += 1
                 if self.warn:
                     msg = repr(e)
                     LOG.warning(msg)
+        if catched_count > 0:
+            types = [exception.__name__ for exception in self.exceptions] \
+                if isinstance(self.exceptions, (list, tuple)) \
+                else self.exceptions.__name__
+            LOG.info(f'CatchExceptionDataset catched {catched_count} exceptions of type {types}.')
```
Comment (thequilo) [inline lazy_dataset/core.py:1789]: Add "of {len(input_dataset)} examples"?

File: lazy_dataset/core.py [hunk 2]
```diff
@@ -1881,9 +1890,14 @@ def catcher(index):
...
                 if data is unique_object:
-                    pass
+                    catched_count += 1
                 else:
                     yield data
+            if catched_count > 0:
+                types = [exception.__name__ for exception in catch_filter_exception] \
+                    if isinstance(catch_filter_exception, (list, tuple)) \
+                    else catch_filter_exception.__name__
+                LOG.info(f'PrefetchDataset catched {catched_count} exceptions of type {types}.')
```
Comment (thequilo) [inline lazy_dataset/core.py:1900]: Add "of {len(input_dataset)} examples"?

File: lazy_dataset/core.py [hunk 3]
```diff
@@ -646,7 +646,9 @@ def filter(self, filter_fn: callable, lazy: bool = True) -> 'Dataset':
                     'You can only use lazy=False if the incoming dataset is '
                     'indexable.'
                 )
-            return self[[i for i, e in enumerate(self) if filter_fn(e)]]
+            idx = [i for i, e in enumerate(self) if filter_fn(e)]
+            LOG.info(f'Filtered {len(self)-len(idx)} from {len(self)} examples.')
```
Comment (thequilo) [inline lazy_dataset/core.py:650]: Do we want to print this message when nothing was filtered? The message in `CatchExceptionDataset` is only printed when an exception was caught.

File: lazy_dataset/core.py (hunk 1 above)
Comment (JanekEbb) [reply lazy_dataset/core.py]: Done

File: lazy_dataset/core.py (hunk 2 above)
Comment (JanekEbb) [reply lazy_dataset/core.py]: Done

File: lazy_dataset/core.py (hunk 3 above)
Comment (JanekEbb) [reply lazy_dataset/core.py]: No, you are right. Fixed now.

**Review by jensheit — APPROVED** [review]

File: lazy_dataset/core.py [hunk 4]
```diff
@@ -1771,14 +1774,22 @@ def __getitem__(self, item):
...
             except self.exceptions as e:
+                catched_count += 1
                 if self.warn:
                     msg = repr(e)
                     LOG.warning(msg)
+        if catched_count > 0:
+            types = ','.join([exception.__name__ for exception in self.exceptions]) \
+                if isinstance(self.exceptions, (list, tuple)) \
+                else self.exceptions.__name__
+            LOG.info(f'CatchExceptionDataset filtered {catched_count} of {total_count} examples (catched expections: {types}).')
```
Comment (boeddeker) [inline lazy_dataset/core.py:1792]: Can you replace `f'CatchExceptionDataset` with `self.__class__.__name__`?

File: lazy_dataset/core.py [hunk 5]
```diff
@@ -1873,17 +1884,25 @@ def catcher(index):
...
                 if data is unique_object:
-                    pass
+                    catched_count += 1
                 else:
                     yield data
+            if catched_count > 0:
+                types = ','.join([exception.__name__ for exception in catch_filter_exception]) \
+                    if isinstance(catch_filter_exception, (list, tuple)) \
+                    else catch_filter_exception.__name__
+                LOG.info(f'PrefetchDataset filtered {catched_count} of {total_count} examples (catched exceptions: {types}).')
```
Comment (boeddeker) [inline lazy_dataset/core.py:1905]: Same here with `PrefetchDataset`

File: lazy_dataset/core.py [hunk 6]
```diff
@@ -2193,9 +2212,16 @@ def __str__(self):
...
+        filtered_count = 0
+        total_count = 0
         for example in self.input_dataset:
+            total_count += 1
             if self.filter_function(example):
                 yield example
+            else:
+                filtered_count += 1
+        if filtered_count > 0:
+            LOG.info(f'FilterDataset filtered {filtered_count} of {total_count} examples.')
```
Comment (boeddeker) [inline lazy_dataset/core.py:2224]: And here

File: lazy_dataset/core.py [hunk 7]
```diff
@@ -1771,14 +1774,22 @@ def __getitem__(self, item):
...
+            total_count += 1
             try:
                 yield input_dataset[i]
             except self.exceptions as e:
+                catched_count += 1
                 if self.warn:
                     msg = repr(e)
                     LOG.warning(msg)
+        if catched_count > 0:
+            types = ','.join([exception.__name__ for exception in self.exceptions]) \
```
Comment (boeddeker) [inline lazy_dataset/core.py:1789]: `','` -> `', '`? i.e. add whitespace

File: lazy_dataset/core.py [hunk 8]
```diff
@@ -2932,7 +2960,7 @@ def __iter__(self):
             else:
                 dropped_count += len(data)
         if dropped_count > 0:
-            print(f'Dropped {dropped_count} examples')
+            LOG.info(f'DynamicBucketDataset dropped {dropped_count} of {total_count} examples.')
```
Comment (boeddeker) [inline lazy_dataset/core.py:2963]: And here with the class name

**Comment by boeddeker** [comment]
Since you replaced one print with logging, could you also do it in `_DiskCacheWrapper`?
Then we probably have no more prints in the code.

**Review by boeddeker — APPROVED** [review]

**Review by thequilo — APPROVED** [review]

**Pull request merged**